src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx

```diff
@@ -1,270 +1,154 @@
-                )}
-                <div className={b('title')}>{getItemTitle(item)}</div>
-            </div>
+            <React.Fragment>
+                {renderList()}
+                <Button view="action" className={applyButtonCn} onClick={onApply}>
+                    {i18n('button_apply')}
+                </Button>
+            </React.Fragment>
         );
     };
 
-    const renderStatus = () => {
-        if (!showStatus) {
-            return null;
-        }
-
-        const selected = getCountSelected();
-        const all = propsItems.length;
-        const status = `${selected}/${all}`;
+    const renderControl: TreeSelectProps<unknown>['renderControl'] = ({toggleOpen}) => {
+        const onKeyDown = createOnKeyDownHandler(toggleOpen);
 
-        return <span className={b('status')}>{status}</span>;
+        return (
+            renderSwitcher?.({onClick: toggleOpen, onKeyDown}) || (
+                <Button onClick={toggleOpen} onKeyDown={onKeyDown}>
+                    <Icon data={Gear} />
+                    {i18n('button_switcher')}
+                </Button>
+            )
+        );
     };
 
-    const renderRequiredColumns = () => {
-        const hasRequiredColumns = requiredItems.length;
+    const onOpenChange = (open: boolean) => {
+        setOpen(open);
 
-        if (!hasRequiredColumns) {
-            return null;
+        if (open === false) {
+            const initialItems = prepareItems(propsItems);
+            setItems(initialItems);
         }
-
-        return (
-            <List
-                items={requiredItems}
-                itemHeight={LIST_ITEM_HEIGHT}
-                itemsHeight={getRequiredListHeight}
-                filterable={filterable}
-                renderItem={renderItem}
-                itemsClassName={b('items')}
-                itemClassName={b('item')}
-                virtualized={false}
-            />
-        );
     };
 
-    const renderConfigurableColumns = () => {
-        return (
-            <List
-                items={currentItems}
-                itemHeight={LIST_ITEM_HEIGHT}
-                itemsHeight={getListHeight}
-                sortable={sortable}
-                filterable={filterable}
-                sortHandleAlign={'right'}
-                onSortEnd={makeOnSortEnd(currentItems)}
-                onItemClick={handleItemClick}
-                renderItem={renderItem}
-                itemsClassName={b('items')}
-                itemClassName={b('item')}
-                virtualized={false}
-            />
-        );
-    };
+    const onUpdate = React.useCallback((selectedItemsIds: string[]) => {
+        setItems((prevItems) => {
+            return prevItems.map((item) => ({
+                ...item,
+                isSelected: selectedItemsIds.includes(item.id),
+            }));
+        });
+    }, []);
 
-    const {onKeyDown: handleControlKeyDown} = useActionHandlers(handleControlClick);
+    const value = React.useMemo(() => {
```

Comment:
1 - do we have strict rules of memoization ?
2 - `map` will allocate new array, then `filter` will allocate another array - and it will happen each rerender.

Reply:
1. We don't. But we use memoization when we know it is really useful, not everywhere.
2. Make sense!